internal/command/docker.go

```diff
@@ -123,6 +140,16 @@ func goModInit(ctx Context) error {
 	return nil
 }
 
+// findGoFlags return the index of context where GOFLAGS is set.
+func findGoFlags(ctx Context) (int, error) {
+	for i, env := range ctx.Env {
+		if strings.Index(env, "GOFLAGS") == 0 {
+			return i, nil
+		}
+	}
+	return -1, errors.New("GOFLAGS not found in context")
+}
```

Comment:
Also, in the same file. Couldn’t we just use strings.Contains?

Reply:
@lucor I did it the "go" way, but yes, it's possible to only return the index that is -1

@Jacalz because I want to be sure that this is the environment variable that is found and nothing more. It is possible that you want to pass "GOFLAGS" as a string in `-X` options (for example to inject a message in a variable at build time to say "with GOFLAGS")

Reply:
@lucor I don't understand your first question. I was able to set multiple `-ldflags` - to set `Version` I also made the test and that worked with: `-e GOFLAGS=-ldflags=-X='main.Version=1.0.0'`

Reply:
> I did it the "go" way, but yes, it's possible to only return the index that is -1 if not found

Thanks for the changes. I see and agree with you about the Go way to return the error. In that specific case I'm ok with a little exception in favour of saving an err check like the `strings.Index` method due the fact it won't return error.

> don't understand your first question. I was able to set multiple -ldflags - to set Version I also made the test and that worked with: -e GOFLAGS=-ldflags=-X='main.Version=1.0.0'

I tested locally with a small fyne app and tried to "inject" to variable values with ldflags but despite the fact I was able to see the flags in the fyne-cross compilation output, it was not set for the app.

Code I've used to test:
```
cat main.go 
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

var Title = "Default"
var Name = "Default"

func main() {
	a := app.New()
	w := a.NewWindow(Title)

	hello := widget.NewLabel("Hi " + Name)
	w.SetContent(container.NewVBox(
		hello,
	))
	w.Resize(fyne.NewSize(200, 200))

	w.ShowAndRun()
}
```

Command used to compile:
```
fyne-cross linux --env GOFLAGS="-ldflags=-X='main.Name=Podman'" --debug .
```

fyne-cross logs:
```
[i] Building binary...
/usr/bin/docker run --rm -t -w /app -v /code/lucor/test:/app:z -v /home/lucor/.cache/fyne-cross:/go:z -e CGO_ENABLED=1 -e GOCACHE=/go/go-build -e GOFLAGS=-ldflags=-X='main.Name=Podman' -ldflags=-w -ldflags=-s -e GOOS=linux -e GOARCH=amd64 -e CC=gcc -e fyne_uid=1000 fyneio/fyne-cross:1.1-base go build -o /app/fyne-cross/bin/linux-amd64/test -v .
[✓] Binary: /code/lucor/test/fyne-cross/bin/linux-amd64/test
[i] Packaging app...
/usr/bin/docker run --rm -t -w /app -v /code/lucor/test:/app:z -e CGO_ENABLED=1 -e GOCACHE=/go/go-build -e fyne_uid=1000 fyneio/fyne-cross:1.1-base /usr/local/bin/fyne version
fyne cli version v2.0.2
/usr/bin/docker run --rm -t -w /app/fyne-cross/tmp/linux-amd64 -v /code/lucor/test:/app:z -v /home/lucor/.cache/fyne-cross:/go:z -e CGO_ENABLED=1 -e GOCACHE=/go/go-build -e GOFLAGS=-ldflags=-X='main.Name=Podman' -ldflags=-w -ldflags=-s -e GOOS=linux -e GOARCH=amd64 -e CC=gcc -e fyne_uid=1000 fyneio/fyne-cross:1.1-base /usr/local/bin/fyne package -os linux -name test -icon /app/fyne-cross/tmp/linux-amd64/Icon.png -appBuild 1 -appVersion 1.0 -executable /app/fyne-cross/bin/linux-amd64/test
[✓] Package: /code/lucor/test/fyne-cross/dist/linux-amd64/test.tar.gz
```

Application output:
![image](https://user-images.githubusercontent.com/313577/117549921-617ce200-b03d-11eb-81e2-21b615629dd1.png)

Please also notes, that fyne-cross exposes also a `-ldflags` so we should have the following command to work too:

```fyne-cross linux -ldflags="-X=main.Name=Podman" --debug .```


Wrt the multilple ldflags, it seems that when you specify them like:
```
GOFLAGS="-ldflags=-X='main.Name=Podman' -ldflags=-s -ldflags=-w" go build
```
or 
```
go build -ldflags=-X='main.Name=Podman' -ldflags=-s -ldflags=-w
```
only the last flag is used (try to compile changing the order)

The following instead work as expected:

```
go build -ldflags "-X=main.Name=Podman -w -s"
```

Reply:
> Thanks again for working on this @metal3d. It is a very nice feature to have and would be nice to see it landed soon. Do you think you could get this rebased on develop? The tests don't want to start and I think the out-of-date branch might partly be the reason.
> 

I'm currently makeing the merges and conflct resolution. I hope to not make regression...

> I finally managed to test this out on my new Fedora 34 installation. I had to install `podman-docker` to even get it started. Would it be possible to get it to only use `podman`? After that, I ran into this error:
> 

This is what I discussed earlier - I wanted to use the "podman" command instead of "docker" with "podman-docker" mackage. Yes, of course, the best is to use the "podman" command IMHO. I can make some changes for this.

> ```
> ~/go/bin/fyne-cross linux -arch amd64,arm64 -app-id com.github.jacalz.wormhole-gui -icon internal/assets/icon/icon-512.png
> [i] Target: linux/amd64
> [i] Cleaning target directories...
> [✓] "bin" dir cleaned: /home/jacob/Git/wormhole-gui/fyne-cross/bin/linux-amd64
> [✓] "dist" dir cleaned: /home/jacob/Git/wormhole-gui/fyne-cross/dist/linux-amd64
> [✓] "temp" dir cleaned: /home/jacob/Git/wormhole-gui/fyne-cross/tmp/linux-amd64
> [i] Checking for go.mod: /home/jacob/Git/wormhole-gui/go.mod
> [✓] go.mod found
> [i] Building binary...
> USE PODMAN
> Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
> Error: error getting default registries to try: short-name resolution enforced but cannot prompt without a TTY
> [✗] exit status 125
> make: *** [Makefile:51: linux] Error 1
> ```

It's been a while I didn't make tests with my fyne-cross changes, I don't remember how I tested 😅 

I will try to make some tests a soon as the confict and merge / rebase are OK

Reply:
OK, I see the problem.

This is the test file:
```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

var Message = "Default"

func main() {
	app := app.New()
	w := app.NewWindow(Message)
	t := widget.NewLabel(Message)

	w.SetContent(t)
	w.ShowAndRun()
}
```

When I build with:
```
../../fyne-cross/fyne-cross linux -ldflags="-X main.Message=42"
```

The error is:
```
go: parsing $GOFLAGS: non-flag "main.Message=42"
```


If I do:
```
# note the "=" after -X
../../fyne-cross/fyne-cross linux -ldflags="-X=main.Message=42"
```

So it builds, but the `main.Message` is not set...

it seems to be related to https://github.com/golang/go/issues/26849

Reply:
OK... I have new information.

The documentation says:

> GOFLAGS
>	A space-separated list of -flag=value settings to apply
>	to go commands by default, when the given flag is known by
>	the current command. Each entry must be a standalone flag.
>	Because the entries are space-separated, flag values must
>	not contain spaces. Flags listed on the command line
>	are applied after this list and therefore override it.

So, "flags value" cannot contain spaces... And `-ldflag=-X main.Value=foo` is badly interpreted. This is the only one ldflag that will not work with `-X=main...`

That means that  this particular ldflag shoud be passed in the command line, not in environment. 

So, what I can try is to pass ldflags to the command line...

Reply:
GOFLAGS seems to be fixed in https://github.com/fyne-io/fyne-cross/pull/42/commits/aa8002daec9555333f2747f24a63c93d1c092489

I didn't tried with "docker". Note that now, the "podman" command is used even if "docker" command is an alias